### ISSUE TITLE
Support public folders not associated with an account

### ISSUE THREAD

**Issue by dawsonbessinger** [ecederstrand/exchangelib#238]
How can one apply the account.calendar.view() function on a public folder calendar?
I am currently getting the public folder and entries by creating a new Folder instance and supplying the folder_id and then filtering, however, this of course does not include the recurring meetings.

**Comment by ecederstrand** [comment]
You can call `view()` on any folder instance, so `Folder(account=a, item_id=..., changekey=...).view()` should work.

**Comment by dawsonbessinger** [comment]
Thanks for the response - I tried your suggestion however I am getting zero items found, no matter how wide of a time span I give it.

My code:
```
public_folder = Calendar(account=account, folder_id=pf_id, changekey=pf_change_key)
items = public_folder.view(
    start=tz.localize(EWSDateTime(2017, 9, 6)),
    end=tz.localize(EWSDateTime(2017, 9, 8)),
)

print(len(items))
for item in items:
	print(item)
```

Also I think the documentation is incorrect:
```
items = account.calendar.view(
    start=tz.localize(EWSDateTime(year, month, day + 1)),
    end=tz.localize(EWSDateTime(year, month, day)),
)
```

If I try this I get the error 'start' must be before 'end'

**Comment by ecederstrand** [comment]
Thanks for the heads-up with the docs. I just corrected this.

`Calendar.view()` definitely works, so something is wrong here. Are you sure the calendar contains valid occurrences between 2017-09-06 and 2017-09-07 23:59:59?

**Comment by dawsonbessinger** [comment]
Okay so if I run this code I get 32 results:
```
public_folder = Folder(account=account, folder_id=pf_id, changekey=pf_change_key)
items = public_folder.filter(
    start__lt=tz.localize(EWSDateTime(2017, 9, 8)),
    end__gt=tz.localize(EWSDateTime(2017, 9, 6)),
)

print(len(items)) # ---> 32
```

If I run this code I get 0 results:
```
public_folder = Calendar(account=account, folder_id=pf_id, changekey=pf_change_key)
items = public_folder.view(
    start=tz.localize(EWSDateTime(2017, 9, 6)),
    end=tz.localize(EWSDateTime(2017, 9, 6) + timedelta(days=2)),
)

print(len(items)) # ---> 0
"""
I am not sure what's wrong but I am ready to try any suggestions you may have!

**Comment by ecederstrand** [comment]
I think I know what's going on. When requesting items, `Calendar` will ignore your `folder_id` and `changekey` values because `Calendar` is a distinguished folder, which produces a different request. It ends up looking in the calendar of `account`, which is not what you wanted.

To access folders of other accounts than your own, you should create a new `Account` object pointing to that account:

```python
a = Account(
    primary_smtp_address=='some.public.account@example.com', 
    credentials=Credentials('your_own_username', 'secret'),
    ...)
for item in a.calendar.view(start=..., end=...):
    print(item)
```

There's a more thorough explanation in https://github.com/ecederstrand/exchangelib#setup-and-connecting

**Comment by ecederstrand** [comment]
@dawsonbessinger Did you get this to work?

**Comment by ecederstrand** [comment]
Closing. Please reopen if you still have issues with this :-)

**Comment by dawsonbessinger** [comment]
Hi,
Sorry for the delay. This public folder calendar is not associated with any account so I am afraid the preferred way you mentioned will not work in this case. I had to switch over to c# - there I only need the corresponding FolderID of the public folder calendar:
```
// Initialize the calendar folder object with only the folder ID. 
CalendarFolder calendar = CalendarFolder.Bind(ews, fid, new PropertySet());
// Set the start and end time and number of appointments to retrieve.
CalendarView cView = new CalendarView(startDate, endDate, MAX_ITEMS);
// Limit the properties returned to the appointment's subject, start time, and end time.
cView.PropertySet = new PropertySet(AppointmentSchema.Subject, AppointmentSchema.Location, AppointmentSchema.Start, AppointmentSchema.End);
// Retrieve a collection of appointments by using the calendar view.
FindItemsResults<Appointment> appointments = calendar.FindAppointments(cView);
```

**Comment by ecederstrand** [comment]
Reopening, since I think we should support folders not associated with an account. I didn't know this was possible.

**Comment by dawsonbessinger** [comment]
This is definitely legacy Exchange behavior on the end-user(s) part, however, I am sure others are in my position where a public folder calendar was created long ago and is now ingrained into their workflow. Thanks for taking another look!

**Comment by ecederstrand** [comment]
I just released v1.10.3. Can you try that version? I tested with a simple:

```python
from exchangelib import Account, Credentials, UTC_NOW
from exchangelib.folders import Calendar

a = Account('some_email', credentials=Credentials('my_email', 'password'), autodiscover=True)

legacy_folder = Calendar(account=a, folder_id="SYAAA=", changekey="oaVoR")

start = UTC_NOW() - timedelta(days=7)
for i in legacy_folder.view(start=start, end=start + timedelta(days=14)):
    print(i)
```

and checked that the request XML for the folder doesn't have a reference to the account. But I don't have access to this kind of folder, so it would be great if you could test.

**Comment by dawsonbessinger** [comment]
I can confirm your changes worked! It pulled all the normal and recurring entries from the public folder calendar in the specified time range. Great job!
